### PR TITLE
Add customize diff for deprecated attribute on gc_policy

### PR DESCRIPTION
**Pull request by ScottSuarez**
Adding customize diff when `google_bigtable_gc_policy.max_age.days` is equivalent to `google_bigtable_gc_policy.max_age.duration`. 

resolves https://github.com/hashicorp/terraform-provider-google/issues/8416

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
bigtable: fixed bug where gc_policy would attempt to recreate the resource when switching from deprecated attribute but maintaining the same value underlying value
```

### CODE REVIEW

**Review by melinath — APPROVED** [review]
This LGTM. Just to make sure I'm understanding, it doesn't look like this tests whether there's a diff, just whether the change succeeds (but without the diff suppression it wouldn't succeed) - correct?

**Review by melinath — CHANGES_REQUESTED** [review]
Looks like tests are failing:

```
Step #24 - "tpg-test": google/resource_bigtable_gc_policy.go:39:14: Error return value of `diff.Clear` is not checked (errcheck)
Step #24 - "tpg-test":      diff.Clear("max_age.0.days")
Step #24 - "tpg-test":                ^
Step #24 - "tpg-test": google/resource_bigtable_gc_policy.go:40:14: Error return value of `diff.Clear` is not checked (errcheck)
Step #24 - "tpg-test":      diff.Clear("max_age.0.duration")
Step #24 - "tpg-test":                ^
Step #23 - "tpgb-test": google-beta/resource_bigtable_gc_policy.go:39:14: Error return value of `diff.Clear` is not checked (errcheck)
Step #23 - "tpgb-test":       diff.Clear("max_age.0.days")
Step #23 - "tpgb-test":                 ^
Step #23 - "tpgb-test": google-beta/resource_bigtable_gc_policy.go:40:14: Error return value of `diff.Clear` is not checked (errcheck)
Step #23 - "tpgb-test":       diff.Clear("max_age.0.duration")
Step #23 - "tpgb-test":                 ^
```

**Comment by melinath** [comment]
I like the idea of unit tests for the diff suppress func, similar to `TestDiskImageDiffSuppress` - would you be open to adding some of those?

File: mmv1/compiler.rb [hunk 1]
```diff
@@ -121,6 +121,7 @@
 end
 
 if override_dir
+  Google::LOGGER.info "Using override directory '#{override_dir}'"
```
[the remark this answers is not in the document]
Comment (ScottSuarez) [reply mmv1/compiler.rb:124]: Yeah sort of a small improvement - although it is a side effect change so I understand if we want to omit

[the remark this answers is not in the document]
Comment (ScottSuarez) [reply mmv1/compiler.rb:124]: yep ! just a small qol improvement for the build process. I can omit if we don't want to piggy back

[the remark this answers is not in the document]
Comment (melinath) [reply mmv1/compiler.rb:124]: I think it's fine, just wanted to make sure it wasn't an accident :-) improvements to logging are great

**Comment by melinath** [comment]
auto-run failed. /gcbrun

File: mmv1/third_party/terraform/utils/test_utils.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func (d *ResourceDataMock) Timeout(key string) time.Duration {
 type ResourceDiffMock struct {
 	Before     map[string]interface{}
 	After      map[string]interface{}
-	Cleared    map[string]struct{}
+	Cleared    map[string]interface{}
```
Comment (melinath) [inline mmv1/third_party/terraform/utils/test_utils.go:77]: I'm surprised this wasn't used anywhere... Could this be `map[string]bool` instead?

**Review by melinath — APPROVED** [review]
LGTM - thanks for the tests!

**Pull request merged**